staging/src/k8s.io/apiserver/pkg/cel/environment/environment.go

```diff
@@ -267,7 +267,10 @@ func (e *EnvSet) filterAndBuildOpts(base *cel.Env, compatVer *version.Version, h
 
 	if len(declTypes) > 0 {
 		provider := apiservercel.NewDeclTypeProvider(declTypes...)
-		providerOpts, err := provider.EnvOptions(base.TypeProvider())
+		if compatVer.AtLeast(version.MajorMinor(1, 31)) {
```

Comment:
if this is pinning to 1,31, how are we able to test creation of CEL API objects making use of this, since master is still at 1.31? seems like we want to prove this works as we want *before* we merge and ship 1.31

Reply:
We tested the creation in multiple places:
The added test in [validation_test](https://github.com/kubernetes/kubernetes/blob/d9900958922b676cb6a4b6241059082a159a004e/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation_test.go#L8876) tested the feature enable/disable in different env
The test in enviroment_test also tested the feature support in different env(for new expr and stored eexpr)
The test in cel/validation_test verified the feature enabled using store version env